### PR TITLE
dirmngr fixes

### PR DESCRIPTION
**Pull request by HarithaIBM**
enabled dirmngr changes to retrieve key from the keyserver

### CODE REVIEW

File: buildenv [hunk 1]
```diff
@@ -40,6 +41,12 @@ echo "cat > $HOME/.gnupg/gpg-agent.conf <<EOF
 pinentry-program $PINENTRY_HOME/bin/pinentry-tty
 log-file \$GPG_CONF_HOME/gpg-agent.log
 EOF"
+
+echo "cat > $HOME/.gnupg/dirmngr.conf <<EOF
```
Comment (IgorTodorovskiIBM) [inline buildenv:45]:
```suggestion
echo "cat > \$HOME/.gnupg/dirmngr.conf <<EOF
```

File: patches/dns-stuff.c.patch [hunk 2]
```diff
@@ -0,0 +1,32 @@
+diff --git a/dirmngr/dns-stuff.c b/dirmngr/dns-stuff.c
+index 0edbc04..0d0b1ed 100644
+--- a/dirmngr/dns-stuff.c
++++ b/dirmngr/dns-stuff.c
+@@ -117,6 +117,10 @@
+ 
+ #define RESOLV_CONF_NAME "/etc/resolv.conf"
+ 
++#if defined (__MVS__) && !defined(NI_MAXHOST
```
Comment (IgorTodorovskiIBM) [inline patches/dns-stuff.c.patch:9]:
```suggestion
+#if defined (__MVS__) && !defined(NI_MAXHOST)
```
Can we remove this with the zoslib change?

File: buildenv (hunk 1 above)
Comment (HarithaIBM) [reply buildenv]: Modified it Thank you.

File: patches/dns-stuff.c.patch (hunk 2 above)
Comment (HarithaIBM) [reply patches/dns-stuff.c.patch]: yes, i have removed. Thank you for review, Igor

File: buildenv [hunk 3]
```diff
@@ -36,10 +37,16 @@ zopen_append_to_setup()
   # echo commands that will run when installing via setup.sh
   echo "GPG_CONF_HOME=\"$HOME/.gnupg\""
```
Comment (IgorTodorovskiIBM) [inline buildenv:38]:
```suggestion
  echo "GPG_CONF_HOME=\"\$HOME/.gnupg\""
```

File: patches/homedir.c.patch [hunk 4]
```diff
@@ -0,0 +1,90 @@
+diff --git a/common/homedir.c b/common/homedir.c
+index 67bbde8..372ba2c 100644
+--- a/common/homedir.c
++++ b/common/homedir.c
+@@ -440,6 +440,63 @@ w32_rootdir (void)
+ 
+ 
+ #ifndef HAVE_W32_SYSTEM /* Unix */
++//__MVS__
```
Comment (IgorTodorovskiIBM) [inline patches/homedir.c.patch:9]:
```suggestion
+//__MVS__
+//TODO: Add to zoslib
```

**Pull request merged**